eng/publish-coverage-results.yml

```diff
@@ -5,18 +5,29 @@ parameters:
   condition: 'succeeded()'
   reports: ''
   assemblyfilters: '-xunit*'
+  classfilters: ''
   breakBuild: false
 
 steps:
-- task: reportgenerator@5
+- task: Powershell@2
   displayName: ReportGenerator
   condition: ${{parameters.condition}}
   inputs:
-    reports: ${{parameters.reports}}
-    targetdir: $(Build.SourcesDirectory)/artifacts/CoverageReport
-    reporttypes: Html;HtmlInline_AzurePipelines_Dark;Cobertura
-    verbosity: 'Verbose'
-    assemblyfilters: ${{parameters.assemblyfilters}}
+    targetType: inline
+    pwsh: true
+    script: |
+      dotnet tool restore
```

Comment:
If we need to restore the tool at every build anyway we could also just install it and wouldn't need `dotnet-tools.json`? Sth. like:
```powershell
dotnet tool install dotnet-reportgenerator-globaltool
dotnet reportgenerator -reports:"${{parameters.reports}}"
```
Unfortunately I can't really test anything related to the Azure DevOps pipeline thus I can't really confirm if it would work.

Reply:
My preferred solution is the Azure DevOps pipeline task (see #1516) because the installation and version update is done automatically. 
The local dotnet tool works in the build pipeline without additional permissions and version update can be done with **dependabot** .  `dotnet tool install -g ...` will fail if the tools was already installed by some other script.